cupy/_core/core.pyx

```diff
@@ -679,12 +679,14 @@ cdef class _ndarray_base:
             raise ValueError(
                 'Changing the dtype of a 0d array is only supported if '
                 'the itemsize is unchanged')
-        if self._c_contiguous:
-            axis = ndim - 1
-        else:
+        axis = ndim - 1
+        if (
+            self._shape[axis] != 1
```

Comment:
The condition to ignore `strides[axis]` is `shape[axis] == 1 or size == 0`.
```suggestion
            self._shape[axis] != 1
            and self.size != 0
```

```
>>> np.ones((3, 20), np.int16)[:0, ::2].view(np.int32)
array([], shape=(0, 5), dtype=int32)
```

Reply:
Thanks! Also added another test to check this.